students/TacSPx/lesson08/circle_class.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python3
+# ---------------------------------------------------------------------------- #
+# Title: circle class
+# Description: Create a class that represents a simple circle.  The circle is
+#              defined by a radius or diameter.  You can query the radius/ diameter,
+#              area, print details, add two circles, compare size, compare equal, and
+#              list/ sort them. Create a Subclass for sphere.
+#
+# <07/25/2020>, Created Script
+#
+# ---------------------------------------------------------------------------- #
+# imports
+import sys
+import math
+
+
+# Step 1
+class Circle(object):
+    def __init__(self, rad):
+        self._rad = rad
+
+    @property
+    def rad(self):
+        return self._rad
+
+    @rad.setter  # radius can't be zero or less
+    def rad(self, val):
+        if val <= 0:  # error handling
+            raise ValueError("Radius can't be zero or a negative number")
+        self._rad = val
+
+    # Step 3
+    @property  # diameter is 2 x radius
+    def dia(self):
+        return 2 * self.rad
+
+    @dia.setter  # diameter is 1/2 the radius
+    def dia(self, dia):
+        self.rad = dia / 2
+
+    # Step 4
+    @property  # area is pi x r2
+    def area(self):
+        return math.pi * self.rad ** 2
+
+    @area.setter
+    def area(self, area):
+        raise AttributeError("Circle can't be directly created with area")
+
+    # Step 5
+    @classmethod
+    def from_diameter(cls, val=0):
+        """Creates circle with 'alternate constructor' using diameter.
+        Class methods take a 'cls' parameter that points to the class and
+        not the object instance"""
+        self = cls(rad=0)
+        self._rad = 0.5 * val
+        self._dia = val
+        return self
```

Comment:
the code should be just a simple line: `return cls(diameter / 2.0)`

you don't need to manually set attributes.